### DISCUSSION TITLE
A GitHub tensorflow API is unavailable

### DISCUSSION THREAD

**Discussion by aimMachinezz** [community/community#38244] [Apps, API and Webhooks] [answered]
https://api.github.com/repos/tensorflow/tensorflow/actions/runs
{
"message": "Server Error"
}

**Comment by LangLangBart** [comment, marked as answer]
I can reproduce the problem, it happens very often with large repos with many branches and PR's.
The server times out after ~10-11 seconds with the following message.

```zsh
❯ gh api -X GET repos/tensorflow/tensorflow/actions/runs
gh: Server Error (HTTP 502)
{
  "message": "Server Error"
}

❯ gh api -X GET repos/microsoft/vscode/actions/runs
gh: Server Error (HTTP 502)
{
  "message": "Server Error"
}
```

--- 

`mislav` gave some advice on this ticket [cli/cli/issues/6560](https://github.com/cli/cli/issues/6560)

```zsh
gh api -X GET repos/tensorflow/tensorflow/actions/runs -f exclude_pull_requests=true -F per_page=10
```

there are more `Query Parameters` in the documents - [list workflow runs for a repository](https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs-for-a-repository)

as you add more filters to the query, the more likely you get results back.

**Reply by aimMachinezz** [reply]
thank you
